### PR TITLE
fix: multiple pool reads parallelize when possible

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: multiple pool reads parallelize when possible

## Motivation and Context
improve performance when more than one pool
is present for lookups on small files.

## How to test this PR?
Nothing special all functionality should work
as is.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#e9c6d566-4959-4df8-9d8b-533e91faefed) |
| mint-erasure.sh | :x: [more...](#ac65f542-2026-4133-9cd1-66826b496651) |
| mint-dist-erasure.sh | :x: [more...](#8c4618d0-c240-471a-914c-8c8a8c00fb53) |
| mint-zoned.sh | :x: [more...](#a145347a-2df4-4464-908e-32d0c890a2a1) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#1fc357d5-33c4-402b-ad3c-ee652879a262) |
<a name="e9c6d566-4959-4df8-9d8b-533e91faefed"></a>
### 11537-fd5549f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30018
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 389ff5d091f6:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 2 minutes and 13 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 56 seconds
(8/15) Running minio-dotnet tests ... done in 47 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 27 seconds
{
  "args": {
    "bucketName": "minio-go-test-aoqqpugtut29bzp5",
    "objectName": "b2b03l0x5eln1jzzpvohekeou6wmti",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 3663,
  "error": "The specified method is not allowed against this resource.",
  "function": "RemoveObjects(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjects",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 1 minutes and 16 seconds
(10/15) Running minio-js tests ... FAILED in 3 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 9,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:684:10) at endReadableNT (internal/streams/readable.js:1327:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(10/15) Running minio-py tests ... done in 3 minutes and 8 seconds
(11/15) Running s3cmd tests ... done in 22 seconds
(12/15) Running s3select tests ... done in 7 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="1fc357d5-33c4-402b-ad3c-ee652879a262"></a>
### 11537-fd5549f/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32147
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2d24c9a11977:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2594,
  "function": "aws --endpoint-url http://minio-dev7.minio.io:32147 s3api copy-object --bucket awscli-mint-test-bucket-12719 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-12719/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 48 seconds
(7/15) Running minio-dotnet tests ... done in 45 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 23 seconds
{
  "args": {
    "bucketName": "minio-go-test-cor5rgq0vk9e4dxb",
    "objectName": "di4ezirdmsr15d2whxve6w522i5zxl",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 3970,
  "error": "The specified method is not allowed against this resource.",
  "function": "RemoveObjects(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjects",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 27 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 128,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-2r1vcc4, objectName = minio-java-test-2id52se-copy, resource = /minio-java-test-2r1vcc4/minio-java-test-2id52se-copy, requestId = 16640C00355302AD, hostId = 6782ab1c-3914-4079-9f47-6c06741cb17d)\nrequest={method=PUT, url=http://minio-dev7.minio.io:32147/minio-java-test-2r1vcc4/minio-java-test-2id52se-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-pg4oaj/minio-java-test-2id52se\nHost: minio-dev7.minio.io:32147\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210215T222241Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210215/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Mon, 15 Feb 2021 22:22:41 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16640C00355302AD\nX-Xss-Protection: 1; mode=block\nDate: Mon, 15 Feb 2021 22:22:41 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 54 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 5 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "3162",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-27699/s3cmd-test-object-32298 cebe1d48-a385-48af-bb3e-d8c9723d4872\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id cebe1d48-a385-48af-bb3e-d8c9723d4872 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 7 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
<a name="a145347a-2df4-4464-908e-32d0c890a2a1"></a>
### 11537-fd5549f/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:32224
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 755db124e314:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 26 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 50 seconds
(8/15) Running minio-dotnet tests ... done in 51 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 39 seconds
{
  "args": {
    "bucketName": "minio-go-test-9j9fmlhud0s30nl0",
    "objectName": "ua5h0950rk4lmaswa3c4v92p03jsor",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 4255,
  "error": "The specified method is not allowed against this resource.",
  "function": "RemoveObjects(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjects",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 2 minutes and 13 seconds
(10/15) Running minio-js tests ... done in 45 seconds
(11/15) Running minio-py tests ... done in 3 minutes and 16 seconds
(12/15) Running s3cmd tests ... done in 19 seconds
(13/15) Running s3select tests ... done in 6 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="8c4618d0-c240-471a-914c-8c8a8c00fb53"></a>
### 11537-fd5549f/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:32170
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 74743fe4e486:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 28 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 55 seconds
(8/15) Running minio-dotnet tests ... done in 51 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 39 seconds
{
  "args": {
    "bucketName": "minio-go-test-cpyivueq9plcsw3j",
    "objectName": "6j0nrifx9csbq5vxa9zh52e4heiroj",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 5189,
  "error": "The specified method is not allowed against this resource.",
  "function": "RemoveObjects(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjects",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 1 minutes and 21 seconds
(10/15) Running minio-js tests ... done in 48 seconds
(11/15) Running minio-py tests ... done in 2 minutes and 58 seconds
(12/15) Running s3cmd tests ... done in 20 seconds
(13/15) Running s3select tests ... done in 6 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="ac65f542-2026-4133-9cd1-66826b496651"></a>
### 11537-fd5549f/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31243
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 43cc7f48661d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 1 seconds
(5/15) Running awscli tests ... done in 2 minutes and 1 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 55 seconds
(8/15) Running minio-dotnet tests ... done in 31 seconds
(9/15) Running minio-go tests ... FAILED in 47 seconds
{
  "args": {
    "bucketName": "minio-go-test-rx5bfefge5uoa1th",
    "objectName": "qqjgjhax4egwp9jmzkxsdfcsg1qk3o",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 6425,
  "error": "The specified method is not allowed against this resource.",
  "function": "RemoveObjects(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjects",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 18 seconds
(10/15) Running minio-js tests ... done in 46 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 22 seconds
(12/15) Running s3cmd tests ... done in 14 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#926c7e7b-43d0-4e41-b85d-995c79015f35) |
<a name="926c7e7b-43d0-4e41-b85d-995c79015f35"></a>
### 11537-c51b613/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:32738
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp eeec72837e0c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2718,
  "function": "aws --endpoint-url http://minio-c2.minio.io:32738 s3api copy-object --bucket awscli-mint-test-bucket-25727 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-25727/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 53 seconds
(7/15) Running minio-dotnet tests ... done in 53 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 14 seconds
{
  "args": {
    "destination": {
      "Bucket": "minio-go-test-jhlaomg6muiqnko6",
      "Object": "dstObject",
      "Encryption": {},
      "UserMetadata": null,
      "ReplaceMetadata": false,
      "UserTags": null,
      "ReplaceTags": false,
      "LegalHold": "",
      "Mode": "",
      "RetainUntilDate": "0001-01-01T00:00:00Z",
      "Size": 0,
      "Progress": null
    },
    "source": {
      "Bucket": "minio-go-test-jhlaomg6muiqnko6",
      "Object": "srcObject",
      "VersionID": "",
      "MatchETag": "",
      "NoMatchETag": "",
      "MatchModifiedSince": "0001-01-01T00:00:00Z",
      "MatchUnmodifiedSince": "0001-01-01T00:00:00Z",
      "MatchRange": false,
      "Start": 0,
      "End": 0,
      "Encryption": null
    }
  },
  "duration": 5040,
  "error": "We encountered an internal error, please try again.: cause(s2: corrupt input)",
  "function": "CopyObject(destination, source)",
  "message": "GetObject failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObject",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 36 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 303,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-31tev6s, objectName = minio-java-test-3ijj7le-copy, resource = /minio-java-test-31tev6s/minio-java-test-3ijj7le-copy, requestId = 16641C4BAC364612, hostId = cac73a33-b013-4921-ab29-b8fba477cfd5)\nrequest={method=PUT, url=http://minio-c2.minio.io:32738/minio-java-test-31tev6s/minio-java-test-3ijj7le-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-10sckkg/minio-java-test-3ijj7le\nHost: minio-c2.minio.io:32738\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210216T032117Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210216/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Tue, 16 Feb 2021 03:21:17 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16641C4BAC364612\nX-Xss-Protection: 1; mode=block\nDate: Tue, 16 Feb 2021 03:21:17 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 53 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 12 seconds
(10/15) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "3247",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-32204/s3cmd-test-object-16152 bf624d17-e50f-441c-a79d-3b422692b916\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id bf624d17-e50f-441c-a79d-3b422692b916 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 7 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-server-pool.go [hunk 1]
```diff
@@ -555,16 +609,49 @@ func (z *erasureServerPools) GetObjectInfo(ctx context.Context, bucket, object s
...
+			// some errors such as MethodNotAllowed for delete marker
+			// should be returned upwards.
+			return objInfos[i], err
 		}
-		return objInfo, nil
 	}
+
+	if found >= 0 {
+		return objInfos[found], nil
+	}
```
Comment (poornas) [inline cmd/erasure-server-pool.go:653]: @harshavardhana, for replication of delete markers, the ReplicationStatus of deletemarker is required from ObjectInfo. Current change will drop the additional info returned from erasure layer and instead return a generic VersionNotFound error below - similar impact for ilm transition.

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/erasure-server-pool.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/erasure-server-pool.go:653]: how does it drop it? @poornas the code is the same as before - whichever is success it shall return-  if there is no success found we return the "MethodNotAllowed" error along with objInfo

Comment (harshavardhana) [reply cmd/erasure-server-pool.go:653]: Check line number 647

Comment (poornas) [reply cmd/erasure-server-pool.go:653]: misunderstood the error code returned for Delete markers,my bad

**Review by poornas — APPROVED** [review]
LGTM

**Review by klauspost — COMMENTED** [review]
https://github.com/minio/minio/blob/b3c56b53fb98db364d05c21020652f27c6e9f825/cmd/erasure-server-pool.go#L253 could also do pools concurrently.

**Comment by harshavardhana** [comment]
@klauspost I didn't do that on purpose, since there are some top layers that have a dependency on the implementation such as bucket replication - will get to it if no regressions were found in the current "read" only part.

**Review by klauspost — APPROVED** [review]
LGTM. We can always look at improvements later.

**Pull request merged**